tests/helpers/intl-test-helper.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React, {ReactElement} from 'react';
+import {createIntl, IntlProvider, IntlShape, injectIntl} from 'react-intl';
+import {mount, shallow, ShallowRendererProps, MountRendererProps} from 'enzyme';
+
+const defaultMessages = require('i18n/en.json');
+const defaultIntl = createIntl({
+    locale: 'en',
+    timeZone: 'Etc/UTC',
+    messages: defaultMessages,
+});
+
+export type IntlInjectedElement = ReactElement<any, ReturnType<typeof injectIntl>>;
+export function isIntlInjectedElement(element: ReactElement): element is IntlInjectedElement {
+    const {type} = element;
+    return typeof type === 'object' && 'WrappedComponent' in type;
+}
+
+interface ShallowWithIntlOptions extends ShallowRendererProps {
+    intl?: IntlShape;
+}
+
+export function shallowWithIntl<T extends IntlInjectedElement>(element: T, options?: ShallowWithIntlOptions) {
+    const {intl = defaultIntl, ...shallowOptions} = options || {};
+    const {locale, defaultLocale, messages} = intl;
+
+    if (!isIntlInjectedElement(element)) {
+        throw new Error('shallowWithIntl() allows only components wrapped by injectIntl() HOC. Use shallow() instead.');
+    }
+
```

Comment:
This helper already safe in type-level, but I added exception explicitly for plain JS.